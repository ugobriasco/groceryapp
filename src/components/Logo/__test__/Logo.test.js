import React from 'react';
import renderer from 'react-test-renderer';

import {Logo, styles} from '../';

describe('Logo', () => {

	it('renders correctly', () => {
	const tree = renderer.create(<Logo />).toJSON();
	expect(tree).toBeTruthy();

	});

	it ('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	describe('customization props', () => {
		it('should customize the logo dimension, proportion, and colors', () => {
			const rendered = renderer.create(
				<Logo 
					imageColor='red'
					customWidth='200'
					backgroundColor='blue'
					ratio='3'
				/>
			);
			expect(rendered).toMatchSnapshot();	
		});
	});

});



