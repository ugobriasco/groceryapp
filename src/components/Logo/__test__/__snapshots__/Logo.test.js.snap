// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo customization props should customize the logo dimension, proportion, and colors 1`] = `
<View
  style={undefined}
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Array [
        undefined,
        Object {
          "height": 200,
          "tintColor": "blue",
          "width": 200,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          undefined,
          Object {
            "tintColor": "red",
            "width": 66.66666666666667,
          },
        ]
      }
    />
  </Image>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={undefined}
  >
    The GroceryApp
  </Text>
</View>
`;
